polkadot-parachain/src/command.rs

```diff
@@ -177,34 +190,33 @@ fn load_spec(id: &str) -> std::result::Result<Box<dyn ChainSpec>, String> {
 		)),
 
 		// -- Fallback (generic chainspec)
-		"" => Box::new(chain_spec::get_chain_spec()),
-
+		"" => {
+			log::warn!("No ChainSpec.id specified, so using default one, based on rococo-parachain runtime");
+			Box::new(chain_spec::default::get_chain_spec())
+		},
 		// -- Loading a specific spec from disk
 		path => {
-			let chain_spec = chain_spec::ChainSpec::from_json_file(path.into())?;
-			match chain_spec.runtime() {
-				Runtime::Statemint => Box::new(
-					chain_spec::statemint::StatemintChainSpec::from_json_file(path.into())?,
-				),
-				Runtime::Statemine => Box::new(
-					chain_spec::statemint::StatemineChainSpec::from_json_file(path.into())?,
-				),
+			let path: PathBuf = path.into();
+			match path.runtime() {
+				Runtime::Statemint =>
+					Box::new(chain_spec::statemint::StatemintChainSpec::from_json_file(path)?),
+				Runtime::Statemine =>
+					Box::new(chain_spec::statemint::StatemineChainSpec::from_json_file(path)?),
 				Runtime::Westmint =>
-					Box::new(chain_spec::statemint::WestmintChainSpec::from_json_file(path.into())?),
-				Runtime::CollectivesPolkadot | Runtime::CollectivesWestend =>
-					Box::new(chain_spec::collectives::CollectivesPolkadotChainSpec::from_json_file(
-						path.into(),
-					)?),
+					Box::new(chain_spec::statemint::WestmintChainSpec::from_json_file(path)?),
+				Runtime::CollectivesPolkadot | Runtime::CollectivesWestend => Box::new(
+					chain_spec::collectives::CollectivesPolkadotChainSpec::from_json_file(path)?,
+				),
 				Runtime::Shell =>
-					Box::new(chain_spec::shell::ShellChainSpec::from_json_file(path.into())?),
+					Box::new(chain_spec::shell::ShellChainSpec::from_json_file(path)?),
 				Runtime::Seedling =>
-					Box::new(chain_spec::seedling::SeedlingChainSpec::from_json_file(path.into())?),
-				Runtime::ContractsRococo => Box::new(
-					chain_spec::contracts::ContractsRococoChainSpec::from_json_file(path.into())?,
-				),
+					Box::new(chain_spec::seedling::SeedlingChainSpec::from_json_file(path)?),
+				Runtime::ContractsRococo =>
+					Box::new(chain_spec::contracts::ContractsRococoChainSpec::from_json_file(path)?),
 				Runtime::Penpal(_para_id) =>
-					Box::new(chain_spec::penpal::PenpalChainSpec::from_json_file(path.into())?),
-				Runtime::Generic => Box::new(chain_spec),
+					Box::new(chain_spec::penpal::PenpalChainSpec::from_json_file(path)?),
+				Runtime::Default =>
+					Box::new(chain_spec::default::DefaultChainSpec::from_json_file(path)?),
```

Comment:
we can box after - that would be neater if the compiler lets us?

Reply:
yes, agree, would be neater, rust does not allow return multiple type from function, afaik the workaround is that dynamic dispatch with Box